airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -518,7 +525,7 @@ def _parse_records(
         stream_state: Mapping[str, Any],
         records_schema: Mapping[str, Any],
         stream_slice: Optional[StreamSlice],
-    ) -> Iterable[StreamData]:
+    ) -> Iterable[Record]:
```

Comment:
this wasn't typed correctly because `_parse_response` returns an iterable of records. This is a more accurate type